### PR TITLE
os/fs/smartfs : Read sector data when it open

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
If CRC is enabled, it uses sector buffer. This buffer always include
information of its header, otherwise data corruption happened

### CODE REVIEW

File: os/fs/smartfs/smartfs_smart.c [hunk 1]
```diff
@@ -300,6 +303,23 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
...
 	sf->byteswritten = 0;
+	
+	if ((sf->bflags & SMARTFS_BFLAG_DIRTY) == 0) {
+		/* When using sector buffering, current sector with its header should
+		 * always be present in sf->buffer. Otherwise data corruption may arise
+		 * when writing.
+		 */
+
+		if (sf->currsector != SMARTFS_ERASEDSTATE_16BIT) {
+			smartfs_setbuffer(&readwrite, sf->currsector, 0, fs->fs_llformat.availbytes, (uint8_t *) sf->buffer);
```
Comment (jsdosa) [inline os/fs/smartfs/smartfs_smart.c:314]: Can you remove the white space behind the type?

File: os/fs/smartfs/smartfs_smart.c [hunk 2]
```diff
@@ -300,6 +303,23 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
...
+
```
Comment (amogh-samsung) [inline os/fs/smartfs/smartfs_smart.c:315]: Extra line?

File: os/fs/smartfs/smartfs_smart.c (hunk 1 above)
Comment (amogh-samsung) [inline os/fs/smartfs/smartfs_smart.c:314]: readwrite is defined under CONFIG_SMARTFS_USE_SECTOR_BUFFER
So this entire new code block should be too.

File: os/fs/smartfs/smartfs_smart.c [hunk 3]
```diff
@@ -300,6 +303,23 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 	sf->curroffset = sizeof(struct smartfs_chain_header_s);
 	sf->currsector = sf->entry.firstsector;
 	sf->byteswritten = 0;
+	
+	if ((sf->bflags & SMARTFS_BFLAG_DIRTY) == 0) {
```
Comment (amogh-samsung) [inline os/fs/smartfs/smartfs_smart.c:307]: sf->bflags and sf->buffer also defined only under CONFIG_SMARTFS_USE_SECTOR_BUFFER

**Comment by amogh-samsung** [comment]
Dependency on https://github.com/Samsung/TizenRT/pull/4376 for smartfs_setbuffer function
4376 has to be merged first.

File: os/fs/smartfs/smartfs_smart.c (hunk 3 above)
Comment (Taejun-Kwon) [reply os/fs/smartfs/smartfs_smart.c:342]: yes.

**Pull request merged**